core/deployment/src/main/java/io/quarkus/deployment/ide/IdeProcessor.java

```diff
@@ -104,7 +104,7 @@ public EffectiveIdeBuildItem effectiveIde(LaunchModeBuildItem launchModeBuildIte
                     }
                     if (matches.size() == 1) {
                         result = matches.get(0);
-                    } else if (matches.size() == 0 && runningIdes.size() > 0) {
+                    } else if ((matches.size() == 0 && runningIdes.size() > 0) || matches.size() > 1) {
```

Comment:
```suggestion
                    } else if (matches.size() > 1 || runningIdes.size() > 0) {
```

Reply:
I think it would be better to short circuit the `runningIdes.size() == 0` case above at the start. Keeping it around is dangerous given we will do a `.next()`.

Reply:
@gsmet good catch